### PR TITLE
[TOOLS-3117] Added attention thresholds to Digest chart / Strategical dashboard chart legends with group selectors

### PR DESCRIPTION
**Pull request by jhonypg**

### CODE REVIEW

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/dashboard-chart-abstract.component.scss [hunk 1]
```diff
@@ -10,3 +13,53 @@ section {
...
+                font-weight: normal;
+                font-family: $obj-font-ptsans;
+
+                &--selector {
+                    margin-top: 8px !important;
+                    font-weight: bold !important;
+                    color: $obj-color-snow !important;
+                }
+
+                &--margin_big {
```
Comment (herbert-venancio) [inline strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/dashboard-chart-abstract.component.scss]: Not working properly

margin_big

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/budget/dashboard-chart-budget.component.ts [hunk 2]
```diff
@@ -9,51 +10,23 @@ import { DashboardChartAbstractComponent } from '../abstract-chart/dashboard-cha
...
-        }, {
-            name: 'Budget - Total',
-            data: this.normalizeChartDataPointSeries(budgetData.totalBudget),
-            type: 'spline',
-            color: '#ff0000',
-            lineWidth: 3
-        }];
-    }
+        const budgetDigestSeries: any[] = DashboardChartBudgetSeriesBuilder.from(this.scenarioChartData.budgetData)
+            .withGroupSelector('Budget Digest', 'budget-digest', true)
```
Comment (herbert-venancio) [inline strategical-dashboard/web-app/src/app/dashboard/detail/chart/budget/dashboard-chart-budget.component.ts]: I don't think it needs extra margin here

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/dashboard-chart-abstract-series.builder.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    protected series: ChartSeriesOptionsBuilder[] = [];
+
+    private groupId: string = undefined;
+    private selectorSeries: ChartSeriesOptionsBuilder = undefined;
+
+    public withGroupSelector(groupName: string, groupId: string, shouldAddExtraMargin: boolean = false): T {
+        this.groupId = groupId;
+
+        this.selectorSeries = new ChartSeriesOptionsBuilder()
```
Comment (herbert-venancio) [inline strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/dashboard-chart-abstract-series.builder.ts]: Or pass extraMargin as parameter:
```typescript
this.selectorSeries = new ChartSeriesOptionsBuilder()
    .withName(groupName)
    .withData([])
    .withColor('transparent')
    .asGroupSelector()
    .withExtraMargin(shouldAddExtraMargin);
```

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/defect/dashboard-chart-defect.component.ts [hunk 4]
```diff
@@ -9,45 +11,20 @@ import { DashboardChartAbstractComponent } from '../abstract-chart/dashboard-cha
...
-            color: 'red',
-            lineWidth: 3
-        }];
-    }
+        const defectDigestSeries: any[] = DashboardChartDefectSeriesBuilder.from(this.scenarioChartData.defectData)
+            .withGroupSelector('Defect Digest', 'defect-digest')
+            .withCurrent()
+            .build();
+
+        const defectThresholdSeries: any[] = DashboardChartScopeSeriesBuilder.from(this.scenarioChartData.scopeData)
```
[the remark this answers is not in the document]
Comment (jhonypg) [reply strategical-dashboard/web-app/src/app/dashboard/detail/chart/defect/dashboard-chart-defect.component.ts]: Done

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/dashboard-chart-abstract-series.builder.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import { ChartSeriesOptions } from './series-options/chart-series-options.model';
+
+export abstract class DashboardChartAbstractSeriesBuilder<T> {
+
+    protected series: ChartSeriesOptionsBuilder[] = [];
+
+    private groupId: string = undefined;
+    private selectorSeries: ChartSeriesOptionsBuilder = undefined;
+
+    public withGroupSelector(groupName: string, groupId: string, shouldAddExtraMargin: boolean = false): T {
```
[the remark this answers is not in the document]
Comment (jhonypg) [reply strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/dashboard-chart-abstract-series.builder.ts]: Done

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/defect/dashboard-chart-defect-series.builder.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            .withType('spline')
+            .withVisible(visible)
+            .withColor('red')
+            .withLineWidth(3);
+
+        this.series.push(total);
+        return this;
+    }
+
+    private normalizeDefectChartDataPointSeries(series: any) {
```
[the remark this answers is not in the document]
Comment (jhonypg) [reply strategical-dashboard/web-app/src/app/dashboard/detail/chart/defect/dashboard-chart-defect-series.builder.ts]: Done

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/series-options/abstract-chart-series-group-option.model.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import { ChartSeriesOptions } from './chart-series-options.model';
+
+export abstract class ChartSeriesGroupOption {
+
+    private static ITEM_CSS_CLASS: string = 'obj-group-option';
+    private static ITEM_LABEL_CSS_CLASS: string = 'obj-group-option__label';
+
+    protected constructor(seriesOptions: any, private groupName: string, classNameModifiers?: string[]) {
+        seriesOptions.className += ChartSeriesGroupOption.addModifierClasses(ChartSeriesGroupOption.ITEM_CSS_CLASS, classNameModifiers);
```
[the remark this answers is not in the document]
Comment (jhonypg) [reply strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/series-options/abstract-chart-series-group-option.model.ts]: Done

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/dashboard-chart-abstract.component.scss (hunk 1 above)
Comment (jhonypg) [reply strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/dashboard-chart-abstract.component.scss]: Done

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/budget/dashboard-chart-budget.component.ts (hunk 2 above)
Comment (jhonypg) [reply strategical-dashboard/web-app/src/app/dashboard/detail/chart/budget/dashboard-chart-budget.component.ts]: Done

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/dashboard-chart-abstract-series.builder.ts (hunk 3 above)
Comment (jhonypg) [reply strategical-dashboard/web-app/src/app/dashboard/detail/chart/abstract-chart/dashboard-chart-abstract-series.builder.ts]: Done

**Pull request merged**